File: backend/app/api/routes/enrichment_review.py
from fastapi import APIRouter, HTTPException, Path
from ...schemas.enrichment_review import ApplyRequest, ApplyResult, OnlineLookupRequest, ReviewResponse, SuggestionUpdate
from ...services import enrichment_review_service as service
router=APIRouter(tags=['enrichment'])
@router.get('/enrichment/review',response_model=ReviewResponse)
async def get_review(): return ReviewResponse(**service.get_review())
@router.post('/enrichment/review/preview-refresh',response_model=ReviewResponse)
async def refresh():
    try:return ReviewResponse(**service.refresh_preview())
    except ValueError as exc: raise HTTPException(status_code=422,detail=str(exc))
@router.patch('/enrichment/review/tracks/{track_id}/suggestions/{suggestion_id}',response_model=ReviewResponse)
async def update(body:SuggestionUpdate,track_id:int=Path(ge=1),suggestion_id:str=Path(min_length=1,max_length=128)):
    try:return ReviewResponse(**service.update_suggestion(track_id,suggestion_id,body.decision,body.note,body.selected_fields))
    except LookupError as exc: raise HTTPException(status_code=404,detail=str(exc))
    except ValueError as exc: raise HTTPException(status_code=422,detail=str(exc))
@router.post('/enrichment/review/apply',response_model=ApplyResult)
async def apply(body:ApplyRequest):
    try:return ApplyResult(**service.apply_selected([item.model_dump() for item in body.items],body.confirm))
    except ValueError as exc: raise HTTPException(status_code=422,detail=str(exc))
@router.post('/enrichment/review/tracks/{track_id}/online-lookup',response_model=ReviewResponse)
async def online_lookup(body:OnlineLookupRequest,track_id:int=Path(ge=1)):
    """Explicit, single-track, bounded lookup against Beets or MusicBrainz. Never triggered automatically."""
    try:return ReviewResponse(**service.online_lookup(track_id,body.source))
    except LookupError as exc: raise HTTPException(status_code=404,detail=str(exc))
    except ValueError as exc: raise HTTPException(status_code=422,detail=str(exc))

File: frontend/src/api/enrichmentReview.ts
import { apiFetch } from './client'
import type { BulkEnrichmentActionResult, BulkEnrichmentSummary, EnrichmentApplyResult, EnrichmentDecision, EnrichmentReview, InboxTrackEnrichmentReview } from '../types/enrichmentReview'
export const fetchEnrichmentReview=()=>apiFetch.get<EnrichmentReview>('/enrichment/review')
export const refreshEnrichmentReview=()=>apiFetch.post<EnrichmentReview>('/enrichment/review/preview-refresh',{})
export const updateEnrichmentSuggestion=(trackId:number,id:string,body:{decision:EnrichmentDecision;note:string;selected_fields:Record<string,string>})=>apiFetch.patch<EnrichmentReview>(`/enrichment/review/tracks/${trackId}/suggestions/${id}`,body)
export const applyEnrichmentSuggestion=(trackId:number,id:string,fields:Record<string,string>)=>apiFetch.post<EnrichmentApplyResult>('/enrichment/review/apply',{confirm:true,items:[{track_id:trackId,suggestion_id:id,fields}]})
export const runOnlineLookup=(trackId:number,source:'beets'|'musicbrainz')=>apiFetch.post<EnrichmentReview>(`/enrichment/review/tracks/${trackId}/online-lookup`,{source})
export const fetchInboxTrackEnrichmentReview=(trackId:number)=>apiFetch.get<InboxTrackEnrichmentReview>(`/workspace/inbox/tracks/${trackId}/enrichment-review`)
export const fetchBulkEnrichmentSummary=(trackIds:number[])=>apiFetch.post<BulkEnrichmentSummary>('/workspace/inbox/enrichment-review/summary',{track_ids:trackIds})
export const acceptSafeEnrichmentSuggestions=(trackIds:number[])=>apiFetch.post<BulkEnrichmentActionResult>('/workspace/inbox/enrichment-review/accept-safe',{track_ids:trackIds,confirm:true})
export const keepCurrentBulkEnrichment=(trackIds:number[])=>apiFetch.post<BulkEnrichmentActionResult>('/workspace/inbox/enrichment-review/keep-current',{track_ids:trackIds,confirm:true})
